codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -168,30 +168,110 @@ private static void encode3to4(
+        return (mediumValue & 0xff0000) | (mediumValue & 0xff00) | (mediumValue & 0xff);
+    }
+
+    private static int toIntLE(int mediumValue) {
+        return (mediumValue & 0xff) << 16 | (mediumValue & 0xff00) | (mediumValue & 0xff0000) >>> 16;
+    }
+
+    private static void encode3to4BigEndian(
+            int inBuff, int numSigBytes, ByteBuf dest, int destOffset, byte[] alphabet) {
+        // Packing bytes into an int to reduce bound and reference count checking.
         switch (numSigBytes) {
-        case 3:
-            dest.setByte(destOffset    , ALPHABET[inBuff >>> 18       ]);
-            dest.setByte(destOffset + 1, ALPHABET[inBuff >>> 12 & 0x3f]);
-            dest.setByte(destOffset + 2, ALPHABET[inBuff >>>  6 & 0x3f]);
-            dest.setByte(destOffset + 3, ALPHABET[inBuff        & 0x3f]);
-            break;
-        case 2:
-            dest.setByte(destOffset    , ALPHABET[inBuff >>> 18       ]);
-            dest.setByte(destOffset + 1, ALPHABET[inBuff >>> 12 & 0x3f]);
-            dest.setByte(destOffset + 2, ALPHABET[inBuff >>> 6  & 0x3f]);
-            dest.setByte(destOffset + 3, EQUALS_SIGN);
-            break;
-        case 1:
-            dest.setByte(destOffset    , ALPHABET[inBuff >>> 18       ]);
-            dest.setByte(destOffset + 1, ALPHABET[inBuff >>> 12 & 0x3f]);
-            dest.setByte(destOffset + 2, EQUALS_SIGN);
-            dest.setByte(destOffset + 3, EQUALS_SIGN);
-            break;
+            case 3:
+                dest.setInt(destOffset, alphabet[inBuff >>> 18       ] << 24 |
+                                        alphabet[inBuff >>> 12 & 0x3f] << 16 |
+                                        alphabet[inBuff >>>  6 & 0x3f] << 8  |
+                                        alphabet[inBuff        & 0x3f]);
+                break;
+            case 2:
+                dest.setInt(destOffset, alphabet[inBuff >>> 18       ] << 24 |
+                                        alphabet[inBuff >>> 12 & 0x3f] << 16 |
+                                        alphabet[inBuff >>> 6  & 0x3f] << 8  |
+                                        EQUALS_SIGN);
+                break;
+            case 1:
+                dest.setInt(destOffset, alphabet[inBuff >>> 18       ] << 24 |
+                                        alphabet[inBuff >>> 12 & 0x3f] << 16 |
+                                        EQUALS_SIGN << 8                     |
+                                        EQUALS_SIGN);
+                break;
+            default:
+                // NOOP
+                break;
+        }
+    }
+
+    private static void encode3to4LittleEndian(
+            int inBuff, int numSigBytes, ByteBuf dest, int destOffset, byte[] alphabet) {
+        // Packing bytes into an int to reduce bound and reference count checking.
+        switch (numSigBytes) {
+            case 3:
+                dest.setInt(destOffset, alphabet[inBuff >>> 18       ]       |
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Bitwise OR of signed byte value computed in io.netty.handler.codec.base64.Base64.encode3to4LittleEndian(int, int, ByteBuf, int, byte[]) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3ABIT_IOR_OF_SIGNED_BYTE)